bundles/org.openhab.binding.teleinfo/src/main/java/org/openhab/binding/teleinfo/internal/serial/TeleinfoSerialControllerHandler.java

```diff
@@ -0,0 +1,199 @@
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, ERROR_UNKNOWN_RETRY_IN_PROGRESS);
+                logger.info("Try to restart Teleinfo receiving...");
+                stopReceivingAndCloseSerialPort();
+                openSerialPortAndStartReceiving();
+            }
+        }, 0, 60, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> keepAliveThreadRef = keepAliveThread;
+        if (keepAliveThreadRef != null) {
+            if (!keepAliveThreadRef.isCancelled())
+                keepAliveThreadRef.cancel(true);
+            keepAliveThread = null;
+        }
+        stopReceivingAndCloseSerialPort();
+
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void onFrameReceived(TeleinfoReceiveThread receiveThread, Frame frame) {
+        updateStatus(ThingStatus.ONLINE);
+        fireOnFrameReceivedEvent(frame);
+    }
+
+    @Override
+    public void onInvalidFrameReceived(TeleinfoReceiveThread receiveThread, InvalidFrameException error) {
+        invalidFrameCounter++;
+        updateState(THING_SERIAL_CONTROLLER_CHANNEL_INVALID_FRAME_COUNTER, new DecimalType(invalidFrameCounter));
+    }
+
+    @Override
+    public void onSerialPortInputStreamIOException(TeleinfoReceiveThread receiveThread, IOException e) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, ERROR_UNKNOWN_RETRY_IN_PROGRESS);
+    }
+
+    @Override
+    public boolean continueOnReadNextFrameTimeoutException(TeleinfoReceiveThread receiveThread, TimeoutException e) {
+        logger.warn("Retry in progress. Next retry in {} seconds...", SERIAL_PORT_DELAY_RETRY_IN_SECONDS);
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, ERROR_UNKNOWN_RETRY_IN_PROGRESS);
+        try {
+            Thread.sleep(SERIAL_PORT_DELAY_RETRY_IN_SECONDS * 1000);
+            return true;
+        } catch (InterruptedException e1) {
+            return false;
+        }
+    }
+
+    private void openSerialPortAndStartReceiving() {
+
+        TeleinfoSerialControllerConfiguration config = getConfigAs(TeleinfoSerialControllerConfiguration.class);
+
+        if (config.serialport.trim().isEmpty()) {
+            logger.error("Teleinfo port is not set.");
+            return;
+        }
+
+        logger.info("Connecting to serial port '{}'...", config.serialport);
+        String currentOwner = null;
+        try {
+            final SerialPortIdentifier portIdentifier = serialPortManager.getIdentifier(config.serialport);
+            logger.debug("portIdentifier = {}", portIdentifier);
+            if (portIdentifier == null) {
+                logger.error("No port identifier for '{}'", config.serialport);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        ERROR_OFFLINE_SERIAL_NOT_FOUND);
```

Comment:
The message parameter of `updateStatus()` is logged by the framework. You could remove the log message.